### ISSUE TITLE
[docs] Document hostElement and the new config.

### ISSUE THREAD

**Issue by ditman** [flutter/flutter#114708]
https://github.com/flutter/flutter/issues/32329 introduces some configurable changes to the flutter behavior, which need to be documented in this page:

* https://docs.flutter.dev/development/platform-integration/web/initialization

The following needs to be documented:

* [x] New `engineInitializer.initializeEngine(configuration)` API.
    * [x] Attributes that can be passed as keys for the `configuration` object.
    * [x] Deprecation of the `window` global variables that used to allow this before.
* [ ] Documentation about the `hostElement` behavior (`null` element vs non-`null`).
* [ ] ??? more?

See the [design doc](https://docs.google.com/document/d/1rFM0qmYds1JkntTruBh8eO72xbT2KoP9JdbizucAAwQ/edit) for some literature, and the related issues/PRs for more info.

**Comment by ditman** [comment]
Another one: https://docs.flutter.dev/development/platform-integration/web/renderers#runtime-configuration

**Comment by ditman** [comment]
@jmagman has suggested this to be documented at https://docs.flutter.dev/development/add-to-app

**Comment by jmagman** [comment]
This feature looks a lot to me like our "add to existing app" feature, which we only document as working for iOS and Android.  Would be great to see this documentation in **Adding to a web app** at the add-to-app docs @ditman linked.

These are the docs we point to when customers say they can't rewrite their entire app in Flutter, but want to incrementally adopt it feature-by-feature.
cc @xster 
_______
<img width="256" alt="Screen Shot 2022-12-15 at 11 48 58 AM" src="https://user-images.githubusercontent.com/682784/207953340-fcf10ec6-6f83-4536-b282-cd2870076672.png">

**Comment by ditman** [comment]
@jmagman you're totally right, in fact this is the URL for the latest design doc regarding this feature:

https://docs.flutter.dev/go/web-add-to-app :)

**Comment by RDRRdevelopment** [comment]
This is another link that isn't mentioned on this bug that could be improved with this new change: https://docs.flutter.dev/deployment/web#embedding-a-flutter-app-into-an-html-page

**Comment by RDRRdevelopment** [comment]
Also David - do you by any change have a draft example of the required code snippet to embed using the new hostElement variable? I'm having difficulty getting it to work.

**Comment by ditman** [comment]
@RDRRdevelopment assuming that you've migrated to the new `flutter.js` bootstrap, and are in the `flutter master` channel; try the following in your `index.html`:

```js
    window.addEventListener('load', function(ev) {
      _flutter.loader.loadEntrypoint({
        serviceWorker: {
          serviceWorkerVersion: serviceWorkerVersion,
        },
        onEntrypointLoaded: async function(engineInitializer) {
          let target = document.querySelector('#my_flutter_target');
          // This is the gist of it!
          let appRunner = await engineInitializer.initializeEngine({
            hostElement: target,
          });
          // End of the gist
          await appRunner.runApp();
        }
      });
    });
```

(The important bit is passing the `hostElement` containing an empty `div` to the `initializeEngine` method)

**Comment by ditman** [comment]
I've documented the new document. I rolled back changes to `hostElement`/add-to-app because those are not making it into the next stable yet (womp womp)

**Comment by ditman** [comment]
Reopening to keep track of the `hostElement` changes when needed.
